### ISSUE TITLE
Time part should not be there if time in search_dates output if time information is not present in text.

### ISSUE THREAD

**Issue by KalakondaKrish** [scrapinghub/dateparser#1162]
I have used search_dates in dateparser.search. The output of search_dates is 00:00:00 when time information is not present.
For example in `I will connect with you on Monday`, the output is `[('on monday', datetime.datetime(2023, 4, 10, 0, 0))]`. 

Is it possible for these cases where time is not present, we return only the date part of the datetime and not return time as 0,0? In postprocessing to remove time when it is 0,0. But, it will create problems when the time is indeed 0,0. 
How to solve this?

**Comment by Gallaecio** [comment]
I don’t think we support that for search_dates that the moment. [We do have something like that for parsing, though](https://dateparser.readthedocs.io/en/latest/usage.html#using-datedataparser).

As a workaround, for now, you could parse search match strings again with DateDataParser:

```python
>>> from dateparser import DateDataParser
>>> parser = DateDataParser()
>>> parser.get_date_data("on monday")
DateData(date_obj=datetime.datetime(2023, 4, 10, 0, 0), period='day', locale='en')
```

The date still has time, but `period` indicates the granularity of the parsed date.
